### PR TITLE
splunk + kafka install for func test and library update for prod sec 

### PR DESCRIPTION
**Pull request by rockb1017**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -22,4 +37,93 @@ jobs:
...
+            # Disable SSL for HEC new-token
+            /opt/splunk/bin/splunk http-event-collector update -uri https://$CI_SPLUNK_HOST:$CI_SPLUNK_PORT -enable-ssl 0 -auth $CI_SPLUNK_USERNAME:$CI_SPLUNK_PASSWORD
+            # Setup Indexes
+            /opt/splunk/bin/splunk add index $CI_INDEX_EVENTS -auth $CI_SPLUNK_USERNAME:$CI_SPLUNK_PASSWORD
+            /opt/splunk/bin/splunk http-event-collector update -uri https://$CI_SPLUNK_HOST:$CI_SPLUNK_PORT -name splunk_hec_token -auth $CI_SPLUNK_USERNAME:$CI_SPLUNK_PASSWORD -index $CI_INDEX_EVENTS -indexes $CI_INDEX_EVENTS            
+            sudo /opt/splunk/bin/splunk restart --accept-license --answer-yes --no-prompt
+      - run:
+          name: Install Kafka
+          command: |
+            cd /tmp && wget wget http://www-us.apache.org/dist/kafka/2.4.0/kafka_2.13-2.4.0.tgz
```
Comment (chaitanyaphalak) [inline .circleci/config.yml]: Please use environment variables for the Kafka version

File: .circleci/config.yml [hunk 2]
```diff
@@ -1,10 +1,25 @@
 version: 2
 jobs:
-  build:
-    docker:
-      - image: circleci/openjdk:8-jdk-browsers
+  build_test:
+    resource_class: large
+    machine:
+      image: ubuntu-1604:201903-01
```
Comment (chaitanyaphalak) [inline .circleci/config.yml]: does this image have jdk8 installed?

Comment (rockb1017) [reply .circleci/config.yml]: yes it does

**Comment by rockb1017** [comment]
iml file is a file that intellij creates for itself. it is in the gitignore file as well. i made changes according to your comments and some more improvements. 
let's have kafka 2.4 pipeline working and replicate it for kafka 1.0 as a separate workload.

**Pull request merged**